parity/rpc.rs

```diff
@@ -253,7 +255,15 @@ pub fn new_ipc<D: rpc_apis::Dependencies>(
 		}
 	}
 
-	match rpc::start_ipc(&conf.socket_addr, handler, rpc::RpcExtractor) {
+	// some validations ..
+	let chmod = conf.chmod;
+	if chmod.len() != 3 && chmod.len() != 4 {
+		return Err("valid octal permission are either 3 or 4 digits long.".into())
+	}
+	let chmod = u16::from_str_radix(&chmod, 8)
```

Comment:
I think `u16::from_str_radix` will catch all errors above except `chmod.len() >  4`

Reply:
Maybe another dumb q but why not do the validation here instead of on the `len()`? Like, `assert!(chmod > 0 && chmod <= 0o777, "no no what even is that?");`?

Reply:
right, [from_str_radix](https://doc.rust-lang.org/std/primitive.u16.html#method.from_str_radix) checks that each digit is `0-7` so should be sufficient with length check. 

However,  it could be an option the perform `chmod > 0 && chmod <= 0o7777` after parsing it as an u16.